src/core/cgroup.c

```diff
@@ -4178,6 +4178,16 @@ int unit_cgroup_freezer_action(Unit *u, FreezerAction action) {
         if (!u->cgroup_realized)
                 return -EBUSY;
 
+        if (action == FREEZER_THAW) {
+                Unit *slice = UNIT_GET_SLICE(u);
+
+                if (slice) {
+                        r = unit_cgroup_freezer_action(slice, FREEZER_THAW);
+                        if (r < 0)
+                                return log_unit_error_errno(u, r, "Failed to thaw slice for unit: %m");
```

Comment:
Maybe, better to include the name of the slice unit in the log.
`log_unit_error_errno(u, r, "Failed to thaw slice %s: %m", slice->id);`
or
`log_unit_error_errno(slice, r, "Failed to thaw the slice unit for %s: %m", u->id);`